### DISCUSSION TITLE
Random author after import

### DISCUSSION THREAD

**Discussion by dumoulina** [community/community#33071] [Apps, API and Webhooks] [answered]
I just have imported a git from gitlab to github and some anonymous commit got a random github account author.
[the imported project on github](https://github.com/Tendocat/openGL/commits/master)
![the former gitlab project](https://user-images.githubusercontent.com/48944658/190221082-6c8e1358-b459-48b3-9769-9a0fe6a5f9d0.png)

**Comment by airtower-luna** [comment, marked as answer]
The connection from author to account is based on the author email address recorded in the commit. That "add textures" commit seems to have an example email address (which apparently someone else added to their account). On the other hand, the "test sur PC fac" account seems to have an address that's not connected to any GitHub account. If it's yours, you can add it to your account and the commit will be recognized as yours. Similar for the other commits.

For future commits, make sure you [set the author email address](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address) in your Git config.

To fix past commits you'd have to [rewrite the repository history](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History), which can cause trouble if you've already shared the repository with others. If you want to do it anyway, there's a "Changing Email Addresses Globally" section at the end of that documentation article that tells you how to do it.
